homeassistant/components/http/auth.py

```diff
@@ -235,4 +282,36 @@ async def auth_middleware(
         request[KEY_AUTHENTICATED] = authenticated
         return await handler(request)
 
+    app.middlewares.append(session_middleware(HomeAssistantCookieStorage(hass)))
     app.middlewares.append(auth_middleware)
+
+
+@callback
+def _async_perform_action_on_non_local(
+    request: Request,
+    strict_connection_static_file_content: str | None,
+) -> StreamResponse | None:
+    """Perform strict connection mode action if the request is not local."""
```

Comment:
I'd add what the actions are in the body of the docstring.

1. Check ip address.
2. If ip address is local let the request through, otherwise 3.
3. Show static page if that's enabled, otherwise 4.
4. Close connection.